std/array.d

```diff
@@ -3661,7 +3672,7 @@ unittest
     {
         return [new shared A].inputRangeObject;
     }
-    auto res = foo.array;
+    static assert(__traits(compiles, foo.array));
```

Comment:
Why are you removing traits compiles in some areas but adding them in others?

shouldn't this be `assert(foo.array == [new shared A]);`?